File: src/services/volunteerService.js
import api from './api';

export const volunteerService = {
  registerVolunteer: async (volunteerDTO) => {
    const response = await api.post('/api/volunteers/register', volunteerDTO);
    return response.data;
  },

  getVolunteerByUserId: async (userId) => {
    const response = await api.get(`/api/volunteers/user/${userId}`);
    return response.data;
  },

  getVolunteerById: async (id) => {
    const response = await api.get(`/api/volunteers/${id}`);
    return response.data;
  },

  getAllVolunteers: async () => {
    const response = await api.get('/api/volunteers');
    return response.data;
  },

  updateVolunteer: async (id, volunteerDTO) => {
    const response = await api.put(`/api/volunteers/${id}`, volunteerDTO);
    return response.data;
  },

  deleteVolunteer: async (id) => {
    const response = await api.delete(`/api/volunteers/${id}`);
    return response.data;
  },

  getVolunteersByCity: async (city) => {
    const response = await api.get(`/api/volunteers/city/${city}`);
    return response.data;
  },

  getVolunteersByStatus: async (status) => {
    const response = await api.get(`/api/volunteers/status/${status}`);
    return response.data;
  },

  getVolunteersByAvailability: async (availability) => {
    const response = await api.get(`/api/volunteers/availability/${availability}`);
    return response.data;
  },

  searchVolunteerByName: async (name) => {
    const response = await api.get(`/api/volunteers/search/${name}`);
    return response.data;
  },
};

export default volunteerService;

File: src/main/java/com/foodrescue/service/VolunteerService.java
package com.foodrescue.service;


import java.util.List;

import com.foodrescue.dto.VolunteerDTO;


public interface VolunteerService {



    // Register Volunteer Profile

    VolunteerDTO registerVolunteer(
            VolunteerDTO volunteerDTO
    );



    // Get Volunteer By ID

    VolunteerDTO getVolunteerById(
            Long volunteerId
    );



    // Get Volunteer Profile By User ID

    VolunteerDTO getVolunteerByUserId(
            Long userId
    );



    // Get All Volunteers

    List<VolunteerDTO> getAllVolunteers();



    // Update Volunteer Profile

    VolunteerDTO updateVolunteer(
            Long volunteerId,
            VolunteerDTO volunteerDTO
    );



    // Delete Volunteer

    void deleteVolunteer(
            Long volunteerId
    );



    // Find By City

    List<VolunteerDTO> getVolunteersByCity(
            String city
    );



    // Find By Status

    List<VolunteerDTO> getVolunteersByStatus(
            String status
    );



    // Find By Availability

    List<VolunteerDTO> getVolunteersByAvailability(
            String availability
    );



    // Search By Name

    List<VolunteerDTO> searchVolunteerByName(
            String fullName
    );


}